imports/plugins/core/accounts/server/no-meteor/mutations/inviteShopMember.js

```diff
@@ -0,0 +1,159 @@
+import _, { difference } from "lodash";
+import SimpleSchema from "simpl-schema";
+import Random from "@reactioncommerce/random";
+import ReactionError from "@reactioncommerce/reaction-error";
+import { Meteor } from "meteor/meteor";
+import { Accounts as MeteorAccounts } from "meteor/accounts-base";
+import Reaction from "/imports/plugins/core/core/server/Reaction";
+import getCurrentUserName from "/imports/plugins/core/accounts/server/no-meteor/util/getCurrentUserName";
+import getDataForEmail from "/imports/plugins/core/accounts/server/util/getDataForEmail";
+
+const inputSchema = new SimpleSchema({
+  email: String,
+  groupId: String,
+  name: String,
+  shopId: String
+});
+
+/**
+ * @name accounts/inviteShopMember
+ * @memberof Mutations/Accounts
+ * @summary Invite new admin users (not consumers) to secure access in the dashboard to permissions
+ * as specified in packages/roles
+ * @param {Object} context - GraphQL execution context
+ * @param {Object} input - Necessary input for mutation. See SimpleSchema.
+ * @param {String} input.shopId - shop to invite user
+ * @param {String} input.groupId - groupId to invite user
+ * @param {String} input.email - email of invitee
+ * @param {String} input.name - name of invitee
+ * @return {Promise<Object>} with boolean of found new account === true || false
+ */
+export default async function inviteShopMember(context, input) {
+  inputSchema.validate(input);
+  const { appEvents, collections, user: userFromContext, userHasPermission, userId: userIdFromContext } = context;
+  const { Accounts, Groups, Shops, users } = collections;
+  const {
+    email,
+    groupId,
+    name,
+    shopId
+  } = input;
+
+  if (!userHasPermission(["reaction-accounts"], shopId)) {
+    throw new ReactionError("access-denied", "Access denied");
+  }
+
+  // we always use primary shop data, so retrieve this shop first with `Reaction` helper,
+  // and only query the `Shops` collection if shopId !== primaryShop._id
+  const primaryShop = Reaction.getPrimaryShop();
+  if (!primaryShop) throw new ReactionError("not-found", "No primary shop found");
+
+  let shop = primaryShop;
+  if (primaryShop._id !== shopId) {
+    shop = await Shops.findOne({ _id: shopId });
+  }
+  if (!shop) throw new ReactionError("not-found", "No shop found");
+
+  const group = await Groups.findOne({ _id: groupId });
+  if (!group) throw new ReactionError("not-found", "No gr found");
+
+  // we don't allow direct invitation of "owners", throw an error if that is the group
+  const ownerGroup = await Groups.findOne({ name: "owner" });
+  if (group._id === ownerGroup._id) {
+    throw new ReactionError("bad-request", "Cannot directly invite owner");
+  }
+
+  // An account can add another account to a group as long as the person adding
```

Comment:
@aldeed I modeled this after some code from `addAccountToGroup`, it seemed a little strange using `user` and `Account`, so hopefully got this correct.

Reply:
I'm not sure if this will work, but you could think of this as two steps:
1. Create user if doesn't already exist
2. Set user's account's group to `groupId`

So this would mean pulling the `context.mutations.addAccountToGroup` call out of the `if` and move it to after the `if`. Always do it.

Then you can maybe remove all of this group fetching and permission checking stuff as long as you remove `isInternalCall: true` from the `addAccountToGroup` call, because then all of that permission checking logic is already there?

Reply:
I was contemplating doing it that way, since the check was already done inside of `addAccountToGroup`, I just wasn't sure if we should remove the `isInternalCall` since technically it would be an internal call from this other function. Since you're suggesting it too, seems like it is probably the better way and I'll switch it.